models/genmodels.go

```diff
@@ -118,11 +120,20 @@ func printInputStruct(g *swagger.Generator, op *spec.Operation) error {
 }
 
 func printInputValidation(g *swagger.Generator, op *spec.Operation) error {
-	capOpID := swagger.Capitalize(op.ID)
-	g.Printf("// Validate returns an error if any of the %sInput parameters don't satisfy the\n",
-		capOpID)
-	g.Printf("// requirements from the swagger yml file.\n")
-	g.Printf("func (i %sInput) Validate() error{\n", capOpID)
+	sspp, paramName := swagger.SingleStringPathParameter(op)
+	if sspp {
+		capOpID := swagger.Capitalize(op.ID)
+		g.Printf("// Validate%sInput returns an error if the input parameter doesn't\n",
+			capOpID)
+		g.Printf("// satisfy the requirements in the swagger yml file.\n")
+		g.Printf("func Validate%sInput(%s string) error{\n", capOpID, paramName)
+	} else {
+		capOpID := swagger.Capitalize(op.ID)
+		g.Printf("// Validate returns an error if any of the %sInput parameters don't satisfy the\n",
+			capOpID)
+		g.Printf("// requirements from the swagger yml file.\n")
+		g.Printf("func (i %sInput) Validate() error{\n", capOpID)
```

Comment:
sorry, wasn't very clear. I meant whether we should we write the function as
```
func Validate%sInput(i %sInput) error
```

so it's like the single string path case